### DISCUSSION TITLE
Embed a drawing area into a GTK window

### DISCUSSION THREAD

**Discussion by inmylo** [hajimehoshi/ebiten#2497] [Q&A]
While learning Ebiten I see that it creates its own OS window and drawing area automatically takes the whole area. However, I would like to use a GTK3/GTK4 toolkit to create my own window with features like HeaderBar, etc; and I wish Ebiten uses only a part of the window, so I'm able to use native UI elements (tabs, buttons, dropdowns, etc.).

Is it possible to achieve that? For example, to build a 2 columns layout with Ebiten canvas on the left side and GTK UI elements on the right side.

**Comment by hajimehoshi** [comment]
> Is it possible to achieve that? 

No it is not possible. Ebitengine just provides an X window for an application and doesn't provide a UI part of a GTK window.

**Reply by inmylo** [reply]
I didn't mean Ebiten should have built-in GTK features. The idea was: I create a GTK3 window with some layout first, in some place insert a GTK DrawingArea and then attach Ebiten to this draving area.

**Reply by hajimehoshi** [reply]
This is what I meant. Unfortunately Ebitengine doesn't provide such a feature.
